src/simplewallet/simplewallet.cpp

```diff
@@ -1470,7 +1470,17 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
       }
     }
     if (m_restoring)
+    {
+      std::string errmsg;
+      uint64_t blockchain_target_height = m_wallet->get_daemon_blockchain_target_height(errmsg);
+      if (errmsg.empty() && m_restore_height > blockchain_target_height){
```

Comment:
Technically, >= (restore_height is a block height, the other is a chain height).